api/v1alpha1/stardoguser_types.go

```diff
@@ -39,6 +42,7 @@ type StardogUserCredentialsSpec struct {
 	// Defaults to .metadata.namespace.
 	Namespace string `json:"namespace,omitempty"`
 	// SecretRef references the v1/Secret name which contains the "username" and "password" keys.
+	// +kubebuilder:validation:Required
 	SecretRef string `json:"secretRef,omitempty"`
 }
 
```

Comment:
Actually thinking about the status, if only ever the Status struct contains conditions and no custom fields, we can consolidate all the status structs and helpers into one

Reply:
That's what I already do. I use the same Status for all APIs. Some of them have more some less but they are have a common set of States